File: tests/python/test_app.py
from flask import Flask
import pytest
from pathlib import Path
from ..app import app


@pytest.fixture
def client():
    app.config['TESTING'] = True
    with app.test_client() as client:
        yield client


def test_hello_endpoint(client):
    response = client.get('/hello')
    assert response.status_code == 200
    assert response.json == {'message': 'Hello, World!'}


def test_data_endpoint(client):
    response = client.get('/data')
    assert response.status_code == 200
    if Path('data.csv').exists():
        assert isinstance(response.json, dict)
    else:
        assert response.json == {'error': 'Dataset not found'}

File: tests/julia/test_app.jl
using Test, HTTP, JSON3, FilePaths
include("/Users/craig/flowtest/app.jl")
@testset "Genie API" begin
response = HTTP.get("http://localhost:8000/hello")
@test response.status == 200
json_data = JSON3.read(response.body)
@test json_data.message == "Hello, World!"
response = HTTP.get("http://localhost:8000/data")
@test response.status == 200
if isfile("/Users/craig/flowtest/data.csv")
json_data = JSON3.read(response.body)
@test haskey(json_data, :columns)
@test length(json_data.columns) == 5
else
@test JSON3.read(response.body) == Dict("error" => "Dataset not found")
end
end